packages/components/src/components/Link/Link.stories.tsx

```diff
@@ -124,6 +124,11 @@ export const _ExternalLink: Story = {
       onCancel: () => console.log('Analytics event: Canceled'),
       onPress: () => console.log('Analytics event: Pressed'),
       onConfirm: () => console.log('Analytics event: Confirmed'),
```

Comment:
What do you think about adding some promises to pass to one of stories to test our `async` logic in `useExternalLink`? Could just be a fake promise function that logs to console after a setTimeout or something. 

```js
function somePromise() {
  return new Promise((resolve) => {
    setTimeout(() => {
      console.log("This message is logged after a 2-second delay");
      resolve();
    }, 2000);
  });
}
```

Reply:
Not sure I follow. Like--adding async analytics events? Seems odd for an analytics event to be async.

The conversion to async was around the fact that Linking.openURL is inherently async*, but for some reason seems to work just fine for the error free path without actually being async. However, to handle the promise rejection correctly, it does require async behavior to wait for the OS to fail to find a suitable way to open the URL. Aside from weird async analytics functions, I don't believe there's anything a Link component consumer could do to otherwise to probe the async adjustments in `useExternalLink` since it only takes the URL string and Alert prompt override strings--more custom behavior (e.g. async function not analytics) from the app would be custom onPress logic and circumvent the hook entirely.

*From [the RN doc](https://reactnative.dev/docs/linking#openurl): The method returns a Promise object. If the user confirms the open dialog or the url automatically opens, the promise is resolved. If the user cancels the open dialog or there are no registered applications for the url, the promise is rejected.